clusterctl/cmd/root.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"flag"
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/apiserver/pkg/util/logs"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+var RootCmd = &cobra.Command{
+	Use:   "clusterctl",
+	Short: "cluster management",
+	Long:  `Simple kubernetes cluster management`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// Do Stuff Here
+		cmd.Help()
+	},
+}
+
+var (
+	kubeConfig string
+)
+
+func Execute() {
+	if err := RootCmd.Execute(); err != nil {
+		glog.Exit(err)
+	}
+}
+
+func parseClusterYaml(file string) (*clusterv1.Cluster, error) {
+	bytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster := &clusterv1.Cluster{}
+	err = yaml.Unmarshal(bytes, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return cluster, nil
+}
+
+func parseMachinesYaml(file string) ([]*clusterv1.Machine, error) {
+	bytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	machines := &clusterv1.MachineList{}
+	err = yaml.Unmarshal(bytes, &machines)
+	if err != nil {
+		return nil, err
+	}
+
+	return util.MachineP(machines.Items), nil
+}
+
+func init() {
+	RootCmd.PersistentFlags().StringVarP(&kubeConfig, "kubeconfig", "k", "", "location for the kubernetes config file. If not provided, $HOME/.kube/config is used")
```

Comment:
Please leave all flags out of the skeleton. They can be filled in as code uses them. This ensures the the flags actually do something when they appear and also ensure they are defined in a way that works and is accurate to the behavior of the code.

For example, I already know this definition of kubeconfig will not be accurate for bootstrapping.

Reply:
Sure.